### ISSUE TITLE
[Docs] MeshKit Error Codes Utility

### ISSUE THREAD

**Issue by leecalcote** [layer5io/meshery#3208]
#### Current State
The MeshKit Error Codes Utility has been explained in-depth a couple of times during community meetings and the MeshKit repo contains a bulleted outline of the utility's behavior, caveats, and usage.

#### Desired State
A section in the https://docs.meshery.io/concepts/architecture can be created for MeshKit and can capture much of those same points.

---

#### Contributor Resources

- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
- [Layer5 Community Slack](http://slack.layer5.io)

**Comment by leecalcote** [comment]
@mgfeller you might consider taking this on.

**Comment by divyanshugit** [comment]
Hey @leecalcote, If @mgfeller is not available can you assign this issue to me?

**Comment by mgfeller** [comment]
Thanks @divyanshugit, this is great! I assign the issue to you. 
Please let me know if you have any questions regarding the error codes utility.

**Comment by mgfeller** [comment]
Hi @divyanshugit any progress here?

**Comment by divyanshugit** [comment]
Yupp, I am working on it. But I started it yesterday as I was busy with completing a project related to the semester.

**Comment by divyanshugit** [comment]
I am hoping that I will be done with this issue by this weekend.

**Comment by mgfeller** [comment]
Ok, cool! No real rush, but would be great to have some progress, if you can make it. Btw, I'll be more or less offline for the next 10 days or so.

**Comment by mgfeller** [comment]
Hi @divyanshugit how is it going? Are you able to find some time to work on this issue? Much appreciated of course.

**Comment by divyanshugit** [comment]
Yes, But I had a confusion which I asked over Slack. Should I create a new page for meshkit error code utils as it has a lot of information and give a brief on the main architecture page?

**Comment by mgfeller** [comment]
Oh, I didn't see that, I took a Slack-vacation :smile: 

Good question. Did you see this page: https://docs.meshery.io/project/contributing-error ? A lot of the details are there. 

As a thought, a diagram in https://docs.meshery.io/concepts/architecture#architecture could summarize just the main points (somewhat graphically like the others), including depicting the flow implemented using CI from the errors in the code to the error code reference page https://docs.meshery.io/reference/error-codes, including how error code placeholders are updated by the utility. A one sentence summary could point to https://docs.meshery.io/project/contributing-error for the details. 

What did you have in mind?

**Comment by divyanshugit** [comment]
I thought, at first I will just give a short summary about meshkit and make a table for more information

**Comment by mgfeller** [comment]
Ok, looking forward to this, and the way you'll present the role of the error code utility, how it is used in CI/CD to get the error info from the components to the docs, and why, which I think is important IMHO.

**Comment by divyanshugit** [comment]
Hi @mgfeller, I'm quite confused cause I'm unable to represent the error code utility in form of a table. I don't know how to make svg diagrams. Can you please suggest me something that can help me out here!! Or should I just create a PR in which I give a short description about the error code utility tool and a link that will redirect the user to the contribution error page?

**Comment by mgfeller** [comment]
@divyanshugit we create diagrams as slide in a Google presentation and then export the slide as svg.
